### PR TITLE
Add spice secrets docs

### PR DESCRIPTION
**Pull request by ewgenius**

### CODE REVIEW

File: spiceaidocs/content/en/reference/Secrets/_index.md [hunk 1]
```diff
@@ -0,0 +1,119 @@
...
+## Secret Stores
+
+The secret store is a location where the `secret` objects are stored.
+
+Spice.ai supports four types of secret stores: `file`, `env`, `kubernetes` and `keyring`. The type of secret store is specified in the `store` field of the `secrets` section in the Spicepod manifest.
+
+### File Secret Store
+
+
+The Spice.ai CLI provides two commands, `spice login` and `spice login dremio`, which are designed to simplify the process of storing credentials. When executed, these commands will automatically persist the corresponding credentials in the file secret store.
```
Comment (lukekim) [inline spiceaidocs/content/en/reference/Secrets/_index.md]: Remove Dremio specific information (that would be provided by the Dremio extension docs).

File: spiceaidocs/content/en/reference/Secrets/_index.md [hunk 2]
```diff
@@ -0,0 +1,119 @@
...
+
+### Environment Secret Store
+
+The `env` store type allows Spice.ai to read secrets from environment variables. The environment variables should be formatted like `SPICED_SECRET_<secret-name>_<secret-value-key>`.
```
Comment (lukekim) [inline spiceaidocs/content/en/reference/Secrets/_index.md]: We should make this `SPICE_SECRET_` in both the runtime and the docs.

**Review by lukekim — DISMISSED** [review]

File: spiceaidocs/content/en/reference/Secrets/_index.md (hunk 1 above)
Comment (ewgenius) [reply spiceaidocs/content/en/reference/Secrets/_index.md]: Removed

File: spiceaidocs/content/en/reference/Secrets/_index.md (hunk 2 above)
Comment (ewgenius) [reply spiceaidocs/content/en/reference/Secrets/_index.md]: Update for spice runtime https://github.com/spiceai/spiceai/pull/799

Comment (ewgenius) [reply spiceaidocs/content/en/reference/Secrets/_index.md]: renamed in docs as well

Comment (ewgenius) [reply spiceaidocs/content/en/reference/Secrets/_index.md]: done

**Review by lukekim — DISMISSED** [review]

File: spiceaidocs/content/en/reference/Secrets/_index.md [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+  name: spiced-account-role
+rules:
+  - apiGroups: [""]
+    resources: ["secrets"]
+    verbs: ["get"]
+```
+
+### Keyring Secret Store
+
+> ⚠️ Keyring store can be enabled by building [spiced](https://github.com/spiceai/spiceai/tree/trunk/bin/spiced) with `keyring-secret-store` cargo feature.
```
Comment (phillipleblanc) [inline spiceaidocs/content/en/reference/Secrets/_index.md]:
```suggestion
> ⚠️ Keyring store can be enabled by building [spiced](https://github.com/spiceai/spiceai/tree/trunk/bin/spiced) with `keyring-secret-store` cargo feature.
```

This should be enabled by default

File: spiceaidocs/content/en/reference/Spicepod/_index.md [hunk 4]
```diff
@@ -24,12 +24,33 @@ The kind of Spicepod manifest. The kind is `Spicepod`.
...
+  store: env
+```
+
 ## `metadata`
 
 An optional `map` of metadata.
 
 **Example**
 
+Pulling secrets from environment variables.
```
Comment (phillipleblanc) [inline spiceaidocs/content/en/reference/Spicepod/_index.md]: This seems out of place?

Comment (ewgenius) [reply spiceaidocs/content/en/reference/Spicepod/_index.md]: Yes, removed

File: spiceaidocs/content/en/reference/Secrets/_index.md (hunk 3 above)
Comment (ewgenius) [reply spiceaidocs/content/en/reference/Secrets/_index.md]: Good point, removed that note. Keyring enabled by default in https://github.com/spiceai/spiceai/pull/800

**Review by phillipleblanc — APPROVED** [review]

**Pull request merged**